src/highdicom/content.py

```diff
@@ -1478,15 +1503,318 @@ def __init__(
+                '2^16 elements.'
+            )
+        elif len_data == 2**16:
+            # Per the standard, this is recorded as 0
+            len_data = 0
+        if lut_data.dtype.type == np.uint8:
+            bits_per_entry = 8
+        elif lut_data.dtype.type == np.uint16:
+            bits_per_entry = 16
+        else:
+            raise ValueError(
+                "Numpy array must have dtype uint8 or uint16."
+            )
+
+        if color.lower() not in ('red', 'green', 'blue'):
+            raise ValueError(
+                'Argument "color" must be either "red", "green", or "blue".'
+            )
+        self._attr_name_prefix = f'{color.title()}PaletteColorLookupTable'
+
+        setattr(self, f'{self._attr_name_prefix}Data', lut_data.tobytes())
+        setattr(
+            self,
+            f'{self._attr_name_prefix}Descriptor',
+            [len_data, int(first_mapped_value), bits_per_entry]
+        )
+
+    @property
+    def lut_data(self) -> np.ndarray:
+        """numpy.ndarray: lookup table data"""
+        if self.bits_per_entry == 8:
+            dtype = np.uint8
+        elif self.bits_per_entry == 16:
+            dtype = np.uint16
+        else:
+            raise RuntimeError("Invalid LUT descriptor.")
+        length = self.number_of_entries
+        data = getattr(self, f'{self._attr_name_prefix}Data')
+        array = np.frombuffer(data, dtype)
+        if len(array) != length:
+            raise RuntimeError(
+                'Length of Lookup Table Data does not match the value '
+                'expected from the Lookup Table Descriptor. '
+                f'Expected {length}, found {len(array)}.'
+            )
+        return array
+
+    @property
+    def number_of_entries(self) -> int:
+        """int: Number of entries in the lookup table."""
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[0])
+
+    @property
+    def first_mapped_value(self) -> int:
+        """int: Pixel value that will be mapped to the first value in the
+        lookup table.
+        """
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[1])
+
+    @property
+    def bits_per_entry(self) -> int:
+        """int: Bits allocated for the lookup table data. 8 or 16."""
+        descriptor = getattr(self, f'{self._attr_name_prefix}Descriptor')
+        return int(descriptor[2])
+
+
+class SegmentedColorLUT(Dataset):
+
+    """Class that abstracts a segmented pixel value palette color lookup table.
+    """
+
+    def __init__(
+        self,
+        first_mapped_value: int,
+        number_of_entries: int,
+        segmented_lut_data: np.ndarray,
+        color: str
+    ):
+        """
+
+        Parameters
+        ----------
+        first_mapped_value: int
+            Pixel value that will be mapped to the first value in the
+            lookup table.
+        number_of_entries: int
```

Comment:
We may be able to deduce it from `segmented_lut_data`. However, we would then need to expand the data first (loop over all entries of the segmented LUT data). It may also be good to let the user specify the intended number of entries in the expanded LUT data and use this information to subsequently check whether the LUT data has been encoded correctly. Thoughts?

Reply:
No strong feelings but I would lean to towards deducing it from the data if we can without the check

Reply:
If we parse it anyways in the constructor, then we can omit the parameter

Reply:
Addressed with b71085db106860137b7dbb0fc023eba44c636963